app/javascript/components/upload/UploadSteps.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+
+import ClusteringForm from './ClusteringForm'
+import ImageForm from './ImageForm'
+import CoordinateLabelForm from './CoordinateLabelForm'
+
+
+function RawCountsUploadForm() {
+  return <span> Raw counts go here</span>
+}
+
+function ProcessedExpressionUploadForm() {
+  return <span> Processed Expresion here</span>
+}
+
+
+
+function MetadataUploadForm() {
+  return <span>Meta-----data</span>
+}
+
+export const STEP_ORDER = ['rawCounts', 'processedExpression', 'metadata', 'clustering', 'images', 'coordinateLabels']
+
+export default {
+  rawCounts: {
+    stepTitle: 'Raw Counts',
+    formComponent: RawCountsUploadForm,
+    fileFilter: file => file.file_type === 'Expression Matrix' && file.expression_file_info?.is_raw_counts
```

Comment:
We need to account for `MM Coordinate Matrix` files, but that means bundling, which I know isn't supported yet.